staging/src/k8s.io/legacy-cloud-providers/aws/aws_loadbalancer.go

```diff
@@ -765,10 +773,14 @@ func (c *Cloud) updateInstanceSecurityGroupsForNLB(lbName string, instances map[
 
 	{
 		clientPorts := sets.Int64{}
+		clientProtocol := "tcp"
```

Comment:
Being a single value for all the ports, won't this misset the security groups if a service exposes some TCP and some UDP services?

Reply:
Oops, that's not yet supported in k8s either, pending https://github.com/kubernetes/enhancements/pull/1438, which will require more work in here, as we'll need to support the TCP_UDP NLB port type when TCP and UDP services are both requesting the same port, e.g., DNS.

Reply:
This feature requires additional changes.